js/content/common.js

```diff
@@ -1172,34 +1210,44 @@ let EarlyAccess = (function(){
 
     let self = {};
 
-    let cache = new Set();
     let imageUrl;
 
-    function checkNodes(selectors, selectorModifier) {
+    async function checkNodes(selectors, selectorModifier) {
         selectorModifier = typeof selectorModifier === "string" ? selectorModifier : "";
+        let appidsMap = new Map();
 
         selectors.forEach(selector => {
```

Comment:
I'd recommend generating just one big selector instead. You can combine selectors with commas.